### PR TITLE
Fix #325: Use general encryptor

### LINKED ISSUE #325: Crypto 3.2: Use high level encryptor and decryptor
Use high level encryptor and decryptor created in wultra/powerauth-crypto#493 This will fix V3.2 protocol support

### PR DESCRIPTION
**Pull request by hvge**
This PR replaces usage of low level `EciesEncryptor` and `EciesDecryptor` with new `ClientEncryptor` interface. There's also fix when in some cases library sent activation ID in application scoped encryption header.

### CODE REVIEW

**Review by banterCZ — APPROVED** [review]
Looks good.

File: powerauth-java-cmd-lib/src/main/java/io/getlime/security/powerauth/lib/cmd/steps/AbstractBaseStep.java [hunk 1]
```diff
@@ -179,44 +196,65 @@ public final JSONObject execute(StepLogger stepLogger, Map<String, Object> conte
...
         ResultStatusObject resultStatusObject = model.getResultStatus();
 
-        EciesEncryptor encryptor;
-        EciesParameters parameters;
+        ClientEncryptor encryptor;
         if (securityContext == null) {
-            parameters = getRequestEciesParameters(stepContext.getModel().getVersion(), associatedData);
-            encryptor = SecurityUtil.createEncryptorForActivationScope(applicationSecret, resultStatusObject, eciesSharedInfo, parameters);
+            final byte[] transportMasterKeyBytes = Base64.getDecoder().decode(resultStatusObject.getTransportMasterKey());
+            encryptor = ENCRYPTOR_FACTORY.getClientEncryptor(
```
Comment (romanstrobl) [inline powerauth-java-cmd-lib/src/main/java/io/getlime/security/powerauth/lib/cmd/steps/AbstractBaseStep.java:217]: This code is a bit hard to read, would you mind splitting it a bit? Also the formatting is off for `applicationKey`.

File: powerauth-java-cmd-lib/src/main/java/io/getlime/security/powerauth/lib/cmd/steps/v3/EncryptStep.java [hunk 2]
```diff
@@ -119,31 +115,32 @@ public StepContext<EncryptStepModel, EciesEncryptedResponse> prepareStepContext(
...
         switch (model.getScope()) {
             case "application" -> {
                 // Prepare ECIES encryptor with sharedInfo1 = /pa/generic/application
-                eciesSharedInfo1 = EciesSharedInfo1.APPLICATION_SCOPE_GENERIC;
-                final byte[] associatedData = EciesUtils.deriveAssociatedData(EciesScope.APPLICATION_SCOPE, model.getVersion().toString(), model.getApplicationKey(), null);
-                parameters = getRequestEciesParameters(model.getVersion(), associatedData);
-                encryptor = ECIES_FACTORY.getEciesEncryptorForApplication((ECPublicKey) model.getMasterPublicKey(),
-                        applicationSecret, eciesSharedInfo1, parameters);
+                encryptorId = EncryptorId.APPLICATION_SCOPE_GENERIC;
+                encryptor = ENCRYPTOR_FACTORY.getClientEncryptor(
```
Comment (romanstrobl) [inline powerauth-java-cmd-lib/src/main/java/io/getlime/security/powerauth/lib/cmd/steps/v3/EncryptStep.java:127]: This code is a bit hard to read, would you mind splitting it a bit?

**Review by romanstrobl — APPROVED** [review]
Perfect. I only have 2 comments related to code style.

**Pull request merged**